### DISCUSSION TITLE
LM Adapter Training

### DISCUSSION THREAD

**Discussion by stefan-it** [adapter-hub/adapters#622] [General]
Hi everybody :hugs: 

I would like to perform experiments with first LM Adapters! So the main intention is, to use an existing multilingual LM (more precisely on of my [hmBERT](https://huggingface.co/hmbert) models) and train a LM Adapter. I would like to adapt new languages or new pretrain corpora for existing languages with e.g. less OCR errors or different domain (books instead of newspapers)...

My question is now about hyper-parameters that are recommended for pretraining on a) small corpora (in range of 1-2GB of text) and b) larger corpora (approx. 30GB of text).

So the [example documentation](https://github.com/adapter-hub/adapters/tree/main/examples/pytorch/language-modeling) uses `seq_bn` as adapter config. Is this config recommended for my use case :thinking: 

Many thanks!

**Comment by hSterz** [comment]
Hello @stefan-it, the documentation uses `seq_bn` because this is the architecture the language adapters were originally introduced in (if you are interested in the details you can check out the [MAD-X paper](https://arxiv.org/pdf/2005.00052.pdf). The simplest way would be to use the config and hyperparameters proposed there.
Other adapter methods like LoRA should also work, but we can not recommend specific configs and hyperparameters.
I hope this helps. Let me know if you have any further questions.

**Reply by lenglaender** [reply]
@stefan-it you can additionally take a look at the appendix of our _Adapters_ paper (https://aclanthology.org/2023.emnlp-demo.13/):
- Figure 3 shows which learning rates work well on average for the different adapter types.
- Figure 2 shows that adapters with lower capacity perform better with higher learning rates.

**Comment by vabatta** [comment]
I would like to add to the conversation, and have some guidance and insights on what to look for when using adapters. I've been experimenting simple NER with distilbert models and it works great given our training and validation corpus, but as soon as I try with adapters, the performance and accuracy massively falls.  
Any advice or guidance would be grateful, as adapters could actually be great for our goal on multiple NER classification.
